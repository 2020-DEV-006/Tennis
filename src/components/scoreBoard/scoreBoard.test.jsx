import React from "react";
import ReactDOM from "react-dom";
import ScoreBoard from "./ScoreBoard";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
configure({ adapter: new Adapter() });

describe("ScoreBoard", () => {
  it("render correctly ScoreBoard component correctly", () => {
    const Scoreboard = renderer.create(<ScoreBoard />).toJSON();
    expect(Scoreboard).toMatchSnapshot();
  });

  it("renders first player correctly", () => {
    const firstPlayer = 0;
    const rendered = renderer.create(<ScoreBoard firstPlayer={firstPlayer} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it("renders second player correctly", () => {
    const secondPlayer = 0;
    const rendered = renderer.create(
      <ScoreBoard secondPlayer={secondPlayer} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  
});
