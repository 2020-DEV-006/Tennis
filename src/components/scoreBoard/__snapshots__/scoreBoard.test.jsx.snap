// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBoard render correctly ScoreBoard component correctly 1`] = `
<div
  className="scoreBoard"
>
  <h3>
    scoreBoard
  </h3>
  <div
    className="playerStyles"
  >
    Player1: 
    <div
      className="scoreStyles"
    >
      0
    </div>
  </div>
  <div
    className="playerStyles"
  >
    Player2: 
    <div
      className="scoreStyles"
    >
      0
    </div>
  </div>
</div>
`;

exports[`ScoreBoard renders first player correctly 1`] = `
<div
  className="scoreBoard"
>
  <h3>
    scoreBoard
  </h3>
  <div
    className="playerStyles"
  >
    Player1: 
    <div
      className="scoreStyles"
    >
      0
    </div>
  </div>
  <div
    className="playerStyles"
  >
    Player2: 
    <div
      className="scoreStyles"
    >
      0
    </div>
  </div>
</div>
`;

exports[`ScoreBoard renders player initial score correctly 1`] = `
<div
  className="scoreBoard"
>
  <h3>
    scoreBoard
  </h3>
  <div
    className="playerStyles"
  >
    Player1: 
    <div
      className="scoreStyles"
    >
      0
    </div>
  </div>
  <div
    className="playerStyles"
  >
    Player2: 
    <div
      className="scoreStyles"
    >
      0
    </div>
  </div>
</div>
`;

exports[`ScoreBoard renders second player correctly 1`] = `
<div
  className="scoreBoard"
>
  <h3>
    scoreBoard
  </h3>
  <div
    className="playerStyles"
  >
    Player1: 
    <div
      className="scoreStyles"
    >
      0
    </div>
  </div>
  <div
    className="playerStyles"
  >
    Player2: 
    <div
      className="scoreStyles"
    >
      0
    </div>
  </div>
</div>
`;
